src/services/doppler-legacy-client.ts

```diff
@@ -183,6 +184,24 @@ export interface DopplerLegacyUserData {
 }
 /* #endregion */
 
+export interface Promotions {
+  title: string;
+  functionality: string;
+  description: string;
+  imageUrl: string;
+  backgroundUrl: string;
+}
+
+export function mapPromotionsJson(json: any): Promotions {
```

Comment:
It is not necessary to export this function.

And, I would move all this mapping stuff inside _getBannerData_.

Reply:
done